pkg/cliplugins/bind/plugin/bind.go

```diff
@@ -0,0 +1,203 @@
+	*base.Options
+	// WorkspacePath refers to the absolute reference to the APIExport workspace
+	// to which the user wants to bind.
+	workspacePath string
+	// Name of the APIExport.
+	apiexportName string
+	// APIExportRef is the argument accepted by the command. It contains the
+	// reference to where APIExport exists. For ex: <absolute_ref_to_workspace>:<apiexport>.
+	APIExportRef string
+	// Name of the APIBinding.
+	APIBindingName string
+}
+
+// NewBindOptions returns new BindOptions.
+func NewBindOptions(streams genericclioptions.IOStreams) *BindOptions {
+	return &BindOptions{
+		Options: base.NewOptions(streams),
+	}
+}
+
+// BindFlags binds fields to cmd's flagset.
+func (b *BindOptions) BindFlags(cmd *cobra.Command) {
+	b.Options.BindFlags(cmd)
+
+	cmd.Flags().StringVar(&b.APIBindingName, "name", b.APIBindingName, "Name of the APIBinding to create.")
+}
+
+// Complete ensures all fields are initialized.
+func (b *BindOptions) Complete(args []string) error {
+	if err := b.Options.Complete(); err != nil {
+		return err
+	}
+
+	if len(args) > 0 {
+		b.APIExportRef = args[0]
+	}
+	return nil
+}
+
+// Validate validates the BindOptions are complete and usable.
+func (b *BindOptions) Validate() error {
+	if b.APIExportRef == "" {
+		return errors.New("`root:ws:apiexport_object` reference to bind is required as an argument")
+	}
+
+	if !strings.HasPrefix(b.APIExportRef, "root") {
+		return fmt.Errorf("fully qualified reference to workspace where APIExport exists is required. The format is `root:<ws>:<apiexport>`")
+	}
+
+	return b.Options.Validate()
+}
+
+// Run creates an apibinding for the user.
+func (b *BindOptions) Run(ctx context.Context) error {
+	config, err := b.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	b.apiexportName, b.workspacePath, err = getAPIExportNameAndWorkspace(b.APIExportRef)
+	if err != nil {
+		return err
+	}
+
+	// if apibindingName is not provided, default it to <apiExportname>-bind.
+	if b.APIBindingName == "" {
+		b.APIBindingName = fmt.Sprintf("%s-bind", b.apiexportName)
+	}
+
+	_, currentClusterName, err := pluginhelpers.ParseClusterURL(config.Host)
+	if err != nil {
+		return fmt.Errorf("current URL %q does not point to cluster workspace", config.Host)
+	}
+
+	binding := &apisv1alpha1.APIBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: b.APIBindingName,
+		},
+		Spec: apisv1alpha1.APIBindingSpec{
+			Reference: apisv1alpha1.ExportReference{
+				Workspace: &apisv1alpha1.WorkspaceExportReference{
+					Path:       b.workspacePath,
+					ExportName: b.apiexportName,
+				},
+			},
+		},
+	}
+
+	kcpclient, err := newKCPClusterClient(config)
+	if err != nil {
+		return err
+	}
+
+	_, err = kcpclient.Cluster(currentClusterName).ApisV1alpha1().APIBindings().Create(ctx, binding, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	// wait till APIBinding is created
+	if err := wait.PollImmediate(time.Millisecond*500, time.Second*5, func() (done bool, err error) {
```

Comment:
this loop is not necessary. The create call above is synchronous. You even get back the object directly as first return value.